src/services/Models.service.tsx

```diff
@@ -225,37 +299,66 @@ loader.setDRACOLoader(dracoLoader);
 export const useGltfLoader = (source: Blob | string): GLTF =>
   suspend(
     async () => {
+      let result;
+
       if (source instanceof Blob) {
-        const buffer = await source.arrayBuffer();
-        return (await loader.parseAsync(buffer, '')) as unknown as GLTF;
+        const url = URL.createObjectURL(source);
+        try {
+          result = await loader.loadAsync(url);
+        } finally {
+          URL.revokeObjectURL(url);
+        }
+      } else {
+        result = await loader.loadAsync(source);
       }
-
-      return loader.loadAsync(source);
+      return result;
     },
     [source],
     { lifespan: 100 }
   );
 
 export const useGltfCachedLoader = (source: Blob | string): GLTF => {
-  const cachedGltf = useRef<Map<string, GLTF>>(new Map<string, GLTF>());
+  const cachedGltf = useRef<GLTF | null>(null);
+  const prevSource = useRef<Blob | string | null>(null);
 
-  return suspend(async (): Promise<GLTF> => {
-    if (cachedGltf.current.has(source as string)) {
-      return cachedGltf.current.get(source as string)!;
-    }
-    let result: GLTF;
-    if (source instanceof Blob) {
-      const buffer = await source.arrayBuffer();
-      result = (await loader.parseAsync(buffer, '')) as GLTF;
-    } else {
-      result = await loader.loadAsync(source);
-    }
+  const scene = suspend(
+    async (): Promise<GLTF> => {
+      if (source === prevSource.current && cachedGltf.current) {
+        return cachedGltf.current;
+      }
 
-    cachedGltf.current.set(source as string, result);
-    return result;
-  }, [source]);
-};
+      let gltf: GLTF;
+      if (source instanceof Blob) {
+        const url = URL.createObjectURL(source);
```

Comment:
The array buffer had potential to leak, so using objectURL instead. Alternatively could find a solution to the array buffer issue.